rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -4120,6 +4238,70 @@ fn system_subnet_does_not_check_for_freezing_threshold_on_allocation_changes() {
         .unwrap();
 }
 
+#[test]
+fn install_does_not_reserve_cycles_when_memory_allocation_is_set() {
+    cycles_reserved_for_app_and_verified_app_subnets(|subnet_type| {
+        const CYCLES: Cycles = Cycles::new(1_000_000_000_000_000);
+        const CAPACITY: u64 = 20_000_000_000;
+        const THRESHOLD: u64 = CAPACITY / 2;
+        const USAGE: u64 = CAPACITY - THRESHOLD;
+
+        let mut test = ExecutionTestBuilder::new()
+            .with_subnet_type(subnet_type)
+            .with_subnet_execution_memory(CAPACITY as i64)
+            .with_subnet_memory_reservation(0)
+            .with_subnet_memory_threshold(THRESHOLD as i64)
+            .build();
+
+        test.create_canister_with_allocation(CYCLES, None, Some(THRESHOLD))
+            .unwrap();
+
+        let canister_id = test
+            .create_canister_with_settings(
+                CYCLES,
+                CanisterSettingsArgsBuilder::new()
+                    .with_memory_allocation(USAGE)
+                    .with_reserved_cycles_limit(CYCLES.get())
+                    .build(),
+            )
+            .unwrap();
+        let reserved_cycles = test
```

Comment:
Could we please also check that the reserved cycles are non-zero (i.e., cycles have indeed been reserved)?

Reply:
Did you add such a check?

Reply:
Yes, sorry, forgot to update. In fact, I added an equality check 🤓

Reply:
> In fact, I added an equality check

if you refer to this
```
        assert_eq!(
            reserved_cycles,
            test.cycles_account_manager().storage_reservation_cycles(
                memory_usage_after,
                &ResourceSaturation::new(subnet_memory_usage, THRESHOLD, CAPACITY),
                test.subnet_size(),
            )
        );
```
then this does not check if the reserved cycles are not zero ,right?

Reply:
Indeed, it's kinda indirect if you inspect the function `storage_reservation_cycles`. I fixed in https://github.com/dfinity/ic/pull/5241 (also fixed it for all other tests where I saw a similar pattern).